### PR TITLE
opentabletdriver: misc improvements to package/module

### PR DESCRIPTION
**Pull request by thiagokokada**
###### Motivation for this change

- In `opentabletdriver` package, use `install` instead of `cp` when possible, making the install process more deterministic (i.e.: ensure that the permissions are correct) and avoiding useless `mkdir` commands
- Add `hardware.opentabletdriver.package` option, allowing using of a custom `OpenTabletDriver` package
- Add missing hooks
- Use `copyDesktopItems` instead of copying it manually

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/X11/opentabletdriver/default.nix [hunk 1]
```diff
@@ -119,24 +116,23 @@ stdenv.mkDerivation rec {
         --set DOTNET_ROOT "${dotnet-netcore}" \
         --suffix LD_LIBRARY_PATH : "${lib.makeLibraryPath runtimeDeps}"
 
-    mkdir -p $out/share/{applications,pixmaps}
-
-    cp -r $src/OpenTabletDriver.UX/Assets/* $out/share/pixmaps
+    mkdir -p $out/lib/OpenTabletDriver
+    cp -rv ./OpenTabletDriver/Configurations $out/lib/OpenTabletDriver
```
Comment (thiagokokada) [inline pkgs/tools/X11/opentabletdriver/default.nix:120]: `install` doesn't support recursive directory copy.

**Comment by thiagokokada** [comment]
@SuperSandro2000 Can I have your review here? Should be a pretty uncontroversial one.

**Comment by kisik21** [comment]
LGTM. I don't think this can break something.

/status needs_merger

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 108131` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>opentabletdriver</li>
  </ul>
</details>

**Pull request merged**